src/com/xilinx/rapidwright/design/DesignTools.java

```diff
@@ -2226,7 +2248,11 @@ public static List<SitePinInst> createMissingSitePinInsts(Design design, Net net
             // Likely net inside encrypted IP, let's see if we can infer anything from existing
             // physical description
             for (SiteInst siteInst : new ArrayList<>(net.getSiteInsts())) {
-                for (String siteWire : new ArrayList<>(siteInst.getSiteWiresFromNet(net))) {
+                Map<Net, List<String>> netSiteWiresMap = siteInstToNetSiteWiresMap != null ?
+                        siteInstToNetSiteWiresMap.get(siteInst) : null;
+                List<String> netSiteWires = netSiteWiresMap != null ?
+                        netSiteWiresMap.get(net) : siteInst.getSiteWiresFromNet(net);
+                for (String siteWire : new ArrayList<>(netSiteWires)) {
```

Comment:
I don't think this needs to a copy anymore (it did in the past), especially if you don't update it below.